### ISSUE TITLE
tf.roll within Lambda layer producing incorrect results when converted to ONNX 

### ISSUE THREAD

**Issue by coltonpeltierSE** [onnx/tensorflow-onnx#1615]
Hi, I've submitted this issue to stack overflow [here](https://stackoverflow.com/questions/68381892/tf2onnx-tensorflow-to-onnx-inconsistent-outputs). But I suspect the issue is likely in the conversion of my tensorflow format to onnx, using tf2onnx (v1.9.1).

I'm experimenting with creating a super simple Tensorflow network (one data processing Lambda layer), and then converting the model to ONNX and verifying the results match when calling the ONNX model from onnxruntime. I'm using Tensorflow v2.5.0. & onnxruntime v1.8.1.

`example_input2 = tf.convert_to_tensor([0,1000,2000,2000,2000,3000,3000,4000],dtype=tf.float32)`

Model definition:

```
inp = keras.layers.Input( type_spec=tf.TensorSpec(shape=[None], dtype=tf.float32))
output = keras.layers.Lambda(lambda x: tf.roll(x,shift=-1,axis=0),name="output") (inp)
model = keras.Model(inp,output,name="pipeline")
```

Then I can feed my example_input2 into the network:

`model.predict(example_input2)`

Which provides the desired output (simply a tf.roll operation):

```
array([1000., 2000., 2000., 2000., 3000., 3000., 4000.,    0.],
  dtype=float32)
```
Great! Now I can save my tensorflow model,

`model.save("OnnxInvestigateData/pipeline2", overwrite=True, include_optimizer=False, save_format='tf')`

Then at the shell, I can convert it to an ONNX format using tf2onnx:

`python -m tf2onnx.convert --opset 14 --saved-model pipeline2 --output pipeline2.onnx`

Then, back in python, I can load the onnx model and try to feed through the same input:

```
sess = rt.InferenceSession("OnnxInvestigateData/pipeline2.onnx", log_verbosity_level=2)
xinput = example_input2.numpy()
sess.run(['output'],{"args_0":xinput})
```

Which provides output that matches the input, and not the desired output (which should have been tf.roll'd by -1):

```
[array([   0., 1000., 2000., 2000., 2000., 3000., 3000., 4000.],
       dtype=float32)]
```
I'm totally at a loss as to why the output here isn't matching when I call model.predict from within python on my original keras model. 

FYI: I created a similar model using tf.add() within a Lambda layer, and it worked exactly as desired.

**Comment by TomWildenhain-Microsoft** [comment]
The problem was with our conversion of roll for negative shift values. #1616 should fix.

**Comment by coltonpeltierSE** [comment]
Thank you!

**Comment by TomWildenhain-Microsoft** [comment]
Cool.  Thanks for the reporting the bug.  Also I'm doing an informal survey of tf2onnx use cases: what is your primary motivation for using tf2onnx, and do you plan on running your model on CPU or GPU (or both)?

**Comment by coltonpeltierSE** [comment]
High level: I work in a line of business with all levels of products (cloud, edge software, and embedded devices). We're looking at ways to standardize our analytics such that the same models can be deployed at all the levels in the system. Onnx looks like a good potential candidate and we're already writing models in Tensorflow. So I'm experimenting with the conversion and performing some proof of concepts on the various levels of the system. Happy to connect via call if you want to discuss a bit more.

**Comment by TomWildenhain-Microsoft** [comment]
One of our PMs (Faith) said she would be interested in talking.  Here's her email: 

![image](https://user-images.githubusercontent.com/67606533/125855550-f302071d-62bd-4e47-a5d7-13f87ed37d25.png)
